cli/pcluster/config/validators.py

```diff
@@ -1423,3 +1411,58 @@ def validate_backup_options(
         fsx_imported_file_chunk_size or fsx_import_path or fsx_export_path or fsx_auto_import_policy
     ) and fsx_automatic_backup_retention_days:
         errors.append("Backups cannot be created on S3-linked file systems")
+
+
+def ebs_volume_size_snapshot_validator(section_key, section_label, pcluster_config):
+    """
+    Validate the following cases.
+
+    The EBS snapshot is in "completed" state if it is specified
+    If user specified the volume size, the volume must be larger than then volume size of the EBS snapshot
+    """
+    errors = []
+    warnings = []
+
+    section = pcluster_config.get_section(section_key, section_label)
+    if section.get_param_value("ebs_snapshot_id"):
+        try:
+            ebs_snapshot_id = section.get_param_value("ebs_snapshot_id")
+            snapshot_response_dict = get_ebs_snapshot_info(ebs_snapshot_id, raise_exceptions=True)
+            # validate that the input volume size is larger than the volume size of the EBS snapshot
+            snapshot_volume_size = snapshot_response_dict.get("VolumeSize")
+            volume_size = section.get_param_value("volume_size")
+            if volume_size < snapshot_volume_size:
+                errors.append("The EBS volume size must not be smaller than the volume size of EBS snapshot.")
+            # validate that the state of ebs snapshot
+            if snapshot_response_dict.get("State") != "completed":
+                warnings.append(
+                    "Snapshot {0} is in state '{1}' not 'completed'".format(
+                        ebs_snapshot_id, snapshot_response_dict.get("State")
+                    )
+                )
+            if volume_size > snapshot_volume_size:
+                warnings.append(
+                    "In order to use the full capacity of the volume, you'll need to manually resize the partition "
+                    "according to this doc: "
+                    "https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/recognize-expanded-volume-linux.html"
```

Comment:
* Let's mention something about why the user might need to manually resize the partition (which is because the volume size is larger than the snapshot's size).
* Did we come to a conclusion on whether or not we should include the URL to the user guide?

Reply:
Added